### PR TITLE
Employee to view PTO - Frontend

### PR DESCRIPTION
**Pull request by onkar-saeloun**
## Notion card
https://www.notion.so/saeloun/Frontend-Each-Miru-user-should-be-able-to-view-their-PTO-balance-at-any-point-of-time-075ebfa98b5b4ca99742055f4d756c53

## Summary

This feature adds ability for employee to view their PTO balance

## Preview

https://www.loom.com/share/a6b6b380c55144449613132a05cf7d14

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested on local machine

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by onkar-saeloun** [comment]
Keeping the PR in draft as this PR has dependency over https://github.com/saeloun/miru-web/pull/944

**Comment by supriya3105** [comment]
@onkar-saeloun Why is this PR still in draft ?

**Comment by supriya3105** [comment]
@onkar-saeloun Also since the backend is not ready for this, lets merge this onto a different branch instead of develop. The backend engineer can pull the changes from this branch

**Comment by onkar-saeloun** [comment]
> @onkar-saeloun Why is this PR still in draft ?

@supriya3105 will raise it only once the Admin PTO PR got merged.

**Comment by onkar-saeloun** [comment]
> @onkar-saeloun Also since the backend is not ready for this, lets merge this onto a different branch instead of develop. The backend engineer can pull the changes from this branch

@supriya3105 I think we can merge this to develop after admin PR gets merged. I won't cause any issues as we will not show anything on frontend till we merge backend.
Merging in other PR may cause lot of conflicts while we raise that PR.

File: app/javascript/src/components/Profile/LeaveBalance/index.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      title={result.name}
+    >
+      {date.getDate()}
+    </button>
+  );
+
+  const tileContent = ({ date }) => {
+    let result;
+    if (
+      (result = holiday.find(o => o.date === dayjs(date).format("DD-MM-YYYY")))
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/LeaveBalance/index.tsx:89]: Please move this else-if logic to switch case and move the conditions inside it to functions where we can call them

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Review by apoorv1316 — APPROVED** [review]

**Comment by supriya3105** [comment]
@onkar-saeloun Once you resolve the comments can you reply back with a done and request review again. The reviewer will then mark the conversation as resolved.

**Comment by onkar-saeloun** [comment]
> @onkar-saeloun Once you resolve the comments can you reply back with a done and request review again. The reviewer will then mark the conversation as resolved.

Yes will do it.

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by supriya3105** [comment]
@onkar-saeloun Please reply to the review comment as done so that the reviewer knows about it. Also please update the branch.

**Comment by onkar-saeloun** [comment]
> @onkar-saeloun Please reply to the review comment as done so that the reviewer knows about it. Also please update the branch.

Yes @supriya3105, I have replied to the comments and asked @prasanthchaduvula to review it.

File: app/javascript/src/components/Profile/LeaveBalance/index.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  );
+
+  const getDateWithType = (holiday, date) =>
+    holiday.find(o => o.date === dayjs(date).format("DD-MM-YYYY"));
+
+  const getCalendarButton = (className, date, result) => (
+    <CalendarButton className={className} date={date} result={result} />
+  );
+
+  const tileContent = ({ date }) => {
```
[the remark this answers is not in the document]
Comment (prasanthchaduvula) [reply app/javascript/src/components/Profile/LeaveBalance/index.tsx:93]: @onkar-saeloun Please make the above requested changes

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**